src/App.tsx

```diff
@@ -1,60 +1,116 @@
+import { useState } from 'react';
 import './App.scss';
+import 'bulma/css/bulma.css';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import { TodoList } from './components/TodoList';
+
+import {
+  findUserByID,
+  getPreparedTodos,
+  getHighestTodoID,
+} from './helpers/helpers';
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(getPreparedTodos);
+  const [title, setTitle] = useState('');
+  const [selectedUserID, setSelectedUserID] = useState(0);
+  const [isTitleEmpty, setIsTitleEmpty] = useState(false);
+  const [isUserIdEmpty, setIsUserIdEmpty] = useState(false);
+
+  const clearForm = () => {
+    setTitle('');
+    setSelectedUserID(0);
+    setIsTitleEmpty(false);
+    setIsUserIdEmpty(false);
+  };
+
+  const handleFormSubmit = () => {
+    if (!title || !selectedUserID) {
+      setIsTitleEmpty(!title);
+      setIsUserIdEmpty(!selectedUserID);
+
+      return;
+    }
+
+    setTodos((prevTodos => {
+      const newTodo = {
+        id: getHighestTodoID(todos),
+        title,
+        completed: false,
+        userId: selectedUserID,
+        user: findUserByID(selectedUserID),
+      };
+
+      return [...prevTodos, newTodo];
+    }));
+
+    clearForm();
+  };
 
-export const App = () => {
   return (
-    <div className="App">
-      <h1>Add todo form</h1>
+    <div className="App section content">
+      <h1 className="title is-1">Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={(event) => {
+          event.preventDefault();
+          handleFormSubmit();
+        }}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            <input
+              type="text"
+              placeholder="Write your title"
+              className="input"
+              data-cy="titleInput"
+              value={title}
+              onChange={(event) => {
+                setIsTitleEmpty(false);
```

Comment:
Why do we need to set this variable to false all the time? Maybe, let's do it when isTitleEmpty is true ?

Reply:
Done